django/db/views.py

```diff
@@ -3058,37 +3137,55 @@ def restricted_image(request, filepath):
     response['Content-Length'] = os.path.getsize(_path)
     return response
 
+def download_qc_attached_file(request, id):
+    qf = None
+    try:
+        qf = QCAttachedFile.objects.get(id=id)
+    except Exception,e:
+        msg = str(('could not find qc attached file object for id', id, e))
+        logger.warn(msg)
+        raise Http404(msg)
+    return _download_file(request,qf)
+
 def download_attached_file(request, id):
+    af = None
+    try:
+        af = AttachedFile.objects.get(id=id)
+    except Exception,e:
+        msg = str(('could not find attached file object for id', id, e))
+        logger.warn(msg)
+        raise Http404(msg)
+    return _download_file(request,af)
+        
+def _download_file(request, file_obj):   
     """                                                                         
     Send a file through Django without loading the whole file into              
     memory at once. The FileWrapper will turn the file object into an           
     iterator for chunks of 8KB.                                                 
     """
-    try:
-        af = AttachedFile.objects.get(id=id)
-        logger.debug(str((
-            'send the attached file:', af, request.user.is_authenticated())))
-        if(af.is_restricted and not request.user.is_authenticated()):
+    try:     
+        if(file_obj.is_restricted and not request.user.is_authenticated()):
             logger.warn(str(('access to restricted file for user is denied',
-                             request.user,af)))
+                             request.user,file_obj)))
             return HttpResponse('Log in required.', status=401)
-
-        _path = os.path.join(settings.STATIC_AUTHENTICATED_FILE_DIR,af.filename)
-        if(af.relative_path):
-            _path = os.path.join(settings.STATIC_AUTHENTICATED_FILE_DIR,
-                                 af.relative_path)
+        
+        _path = settings.STATIC_AUTHENTICATED_FILE_DIR
+        if(file_obj.relative_path):
+            _path = os.path.join(_path,file_obj.relative_path)
+        _path = os.path.join(_path,file_obj.filename)
         _file = file(_path)
-        logger.debug(str(('download_attached_file',_path,_file)))
+        logger.info(str(('download_attached_file',_path,_file)))
         wrapper = FileWrapper(_file)
         # use the same type for all files
         response = HttpResponse(wrapper, content_type='text/plain') 
```

Comment:
Potentially application/octet-stream might be more appropriate if the type is unknown. Long term, ideally use the case specific file type if it is possible to determine